AGENTS.md

```diff
@@ -0,0 +1,85 @@
+
+## High-Level Architecture
+
+**Core Framework Stack:**
+- **Fastify** - Web framework with plugins for CORS, helmet, cookies, rate limiting
+- **Mercurius** - GraphQL server with caching, upload support, and subscriptions
+- **TypeORM** - Database ORM with entity-based modeling and migrations
+- **PostgreSQL** - Primary database with master/slave replication setup
+- **Redis** - Caching and pub/sub via ioRedisPool
+- **Temporal** - Workflow orchestration for background jobs
+- **ClickHouse** - Analytics and metrics storage
+
+**Application Entry Points:**
+- `src/index.ts` - Main Fastify server setup with GraphQL, auth, and middleware
+- `bin/cli.ts` - CLI dispatcher supporting api, background, temporal, cron modes
+- `src/background.ts` - Pub/Sub message handlers and background processing
+- `src/cron.ts` - Scheduled task execution
+
+**GraphQL Schema Organization:**
+- `src/graphql.ts` - Combines all schema modules with transformers and directives
+- `src/schema/` - GraphQL resolvers organized by domain (posts, users, feeds, etc.)
+- `src/directive/` - Custom GraphQL directives for auth, rate limiting, URL processing
+- **Docs**: See `src/graphorm/AGENTS.md` for comprehensive guide on using GraphORM to solve N+1 queries. GraphORM is the default and preferred method for all GraphQL query responses. Use GraphORM instead of TypeORM repositories for GraphQL resolvers to prevent N+1 queries and enforce best practices.
+
+**Data Layer:**
+- `src/entity/` - TypeORM entities defining database schema
+- `src/migration/` - Database migrations for schema evolution  
+- `src/data-source.ts` - Database connection with replication configuration
+
+**Core Services:**
+- `src/Context.ts` - Request context with user, permissions, and data loaders
+- `src/auth.ts` - Authentication middleware and user context resolution
+- `src/dataLoaderService.ts` - Efficient batch loading for related entities
+- `src/workers/` - Use workers for async, non-critical operations (notifications, reputation, external syncs). Prefer `TypedWorker` for type safety. Architecture uses Google Pub/Sub + CDC (Debezium) for reactive processing. See `src/workers/AGENTS.md` for more.
+- `src/integrations/` - External service integrations (Slack, SendGrid, etc.) 
+- `src/cron/` - Scheduled cron jobs for maintenance and periodic tasks. One file per cron, registered in `index.ts`, deployed via `.infra/crons.ts` Pulumi config. Each cron exports a `Cron` object with `name` and `handler(DataSource, logger, pubsub)`. Run locally with `pnpm run cli cron <name>`. See `src/cron/AGENTS.md` for more.
+
+**Business Domains:**
+- **Content**: Posts, comments, bookmarks, feeds, sources
+- **Users**: Authentication, preferences, profiles, user experience
+- **Organizations**: Squad management, member roles, campaigns
+- **Notifications**: Push notifications, email digests, alerts
+- **Monetization**: Paddle subscription management, premium features
+
+**Testing Strategy:**
```

Comment:
There's also singular e2e test, but agent might be able to extend if it feels needed